zion/src/test/scala/edu/uci/ics/cloudberry/zion/model/impl/AQLGeneratorTest.scala

```diff
@@ -530,6 +530,70 @@ class AQLGeneratorTest extends Specification {
           |}""".stripMargin.trim)
     }
 
+    "translate lookup one table with one join key" in {
+      val populationDataSet = PopulationDataStore.DatasetName
+      val populationSchema = PopulationDataStore.PopulationSchema
+
+      val selectValues = Seq("*", "population")
+      val selectStatement = SelectStatement(Seq.empty, 0, 0, selectValues)
+      val lookup = LookupStatement(Seq("geo_tag.stateID"), populationDataSet, Seq("stateID"), selectValues,
+        selectValues)
+      val filter = Seq(textFilter)
+      val query = new Query(TwitterDataSet, Seq(lookup), filter, Seq.empty, select = Some(selectStatement))
+      val result = parser.generate(query, Map(TwitterDataSet -> schema, populationDataSet -> populationSchema))
+      removeEmptyLine(result) must_== unifyNewLine(
+        """
+          |for $t in dataset twitter.ds_tweet
+          |for $l0 in dataset twitter.US_population
```

Comment:
definitely not correct. but we have to swap the order, otherwise the performance will be really really bad. 
I think we need to split the `for` part and the `where` part, and return both strings. then put the `for $t in dataset twitter.ds_tweet` in the middle. 

Otherwise, we need to change AsterixDB 😄